### PR TITLE
 collapse onto upstream version command

### PR DESCRIPTION
**Pull request by deads2k**
In 3.11, `oc version` communicates with endpoints on joint apiserver process.  Those endpoints are going away and because of API aggregation, the idea that a particular kube-apiserver maps to a particular openshift-apiserver is no longer valid.  This fixes the `oc version` command to exactly match the upstream behavior.

/assign @soltysh @juanvallejo 
/cc @sttts @jhadvig

### CODE REVIEW

**Comment by sttts** [comment]
"exactly match the upstream behavior." means that we see 1.11? Or that we see 3.11?

**Comment by deads2k** [comment]
> "exactly match the upstream behavior." means that we see 1.11? Or that we see 3.11?

exactly except I allow for injecting a client version.

**Comment by sttts** [comment]
Sgtm.

**Comment by soltysh** [comment]
/hold 
for the rebase to land

**Comment by deads2k** [comment]
the format changes to match upstream.

File: test/cmd/basicresources.sh [hunk 1]
```diff
@@ -29,10 +29,8 @@ os_git_regex="$( escape_regex "${OS_GIT_VERSION%%-*}" )"
 kube_git_regex="$( escape_regex "${KUBE_GIT_VERSION%%-*}" )"
 etcd_version="$(echo "${ETCD_GIT_VERSION}" | sed -E "s/\-.*//g" | sed -E "s/v//")"
 etcd_git_regex="$( escape_regex "${etcd_version%%-*}" )"
-os::cmd::expect_success_and_text 'oc version' "oc ${os_git_regex}"
-os::cmd::expect_success_and_text 'oc version' "kubernetes ${kube_git_regex}"
-os::cmd::expect_success_and_text 'oc version' "features: Basic-Auth"
-os::cmd::expect_success_and_text 'openshift version' "openshift ${os_git_regex}"
+os::cmd::expect_success_and_text 'oc version' "Client Version: .*GitVersion${os_git_regex}"
```
Comment (soltysh) [inline test/cmd/basicresources.sh]: This needs to be:
```
os::cmd::expect_success_and_text 'oc version' "Client Version: .*GitVersion:\"${os_git_regex}"
```
notice the `:\"` between `GitVersion` and `${os_git_regex}`.

**Comment by deads2k** [comment]
network

/retest

**Comment by deads2k** [comment]
serial is a known flake.

**Pull request merged**